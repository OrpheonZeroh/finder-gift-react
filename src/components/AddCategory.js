import React, { useState } from 'react'
import PropTypes from 'prop-types'

export const AddCategory = ({ setCategories } ) => {

    const [inputValue, setinputValue] = useState('') //si lo dejas en blanco, queda como indefinido

    const handleInputChange = (e) => {
        setinputValue(e.target.value)
    }

    const handleSubmit = (e)=> {

        e.preventDefault();

        if(inputValue.trim().length > 2) {
            setCategories(cats=> [inputValue, ...cats]);
            setinputValue('')
        }

        // console.log('Submit hecho')
        

    }

    return (
        <form onSubmit={handleSubmit}>
            <input
                type="text"
                value={ inputValue }
                onChange={handleInputChange}
            />
        </form>
    )

}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}