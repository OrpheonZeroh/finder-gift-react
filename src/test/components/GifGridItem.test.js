import React from 'react'
import {shallow} from 'enzyme'
import { GifGridItem } from '../../components/GifGridItem';


describe('Pruebas con <GifGridItem/>', () => {

    const title = 'Titulo';
    const url = 'http://localhost:8900/epale'
    const wrapper = shallow(<GifGridItem title={title} url={url} />)

    test('debe de mostrar el componente correctamente', () => {

        expect(wrapper).toMatchSnapshot()
        
    });

    test('debe de tener un parrafo con el titulo', () => {
        
        const p = wrapper.find('p')
        expect(p.text().trim()).toBe(title)

    })
    
    test('debe tener la imagen igual al url y alt a los props', () => {
        
        const img = wrapper.find('img');
        // console.log(img.props())console.log(img.prop('src'))
        expect(img.prop('src') ).toBe(url)
        expect(img.prop('alt') ).toBe(title)

    });

    test('debe tener animate__pulse', () => {

        const div = wrapper.find('div')
        const className = (div.prop('className'))
        //expect(div.prop('div') ).toBe(css_class)
        expect(className.includes('animate__pulse')).toBe(true)
        
    })
    

    
})
