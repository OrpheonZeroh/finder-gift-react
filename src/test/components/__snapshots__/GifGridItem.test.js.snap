// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas con <GifGridItem/> debe de mostrar el componente correctamente 1`] = `
<div
  className="card animate__animated animate__pulse"
>
  <img
    alt="Titulo"
    src="http://localhost:8900/epale"
  />
  <p>
    Titulo
  </p>
</div>
`;
